sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -0,0 +1,919 @@
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptID;
+import org.apache.hadoop.mapreduce.task.TaskAttemptContextImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+
+/**
+ * A bounded source for any Data source which implements HadoopInputFormat. For E.g. Cassandra,
+ * Elastic Search, HBase, Redis , PostGres etc.
+ * <p>
+ * The HadoopInputFormat source returns a set of {@link org.apache.beam.sdk.values.KV} key-value
+ * pairs returning a {@code PCollection<KV>}.
+ * 
+ * 
+ * <p>
+ * HadoopConfiguration is mandatory for reading the data using HadoopInputFormat source.
+ * <p>
+ * {@link org.apache.hadoop.conf.Configuration HadoopConfiguration} object will have to be set with
+ * following properties without fail:
+ * <ul>
+ * <li>mapreduce.job.inputformat.class</li>
+ * <li>key.class</li>
+ * <li>value.class</li>
+ * </ul>
+ * For example:
+ * 
+ * <pre>
+ * <tt>
+ *   Configuration myHadoopConfiguration = new Configuration(false);
+ *   // Set Hadoop InputFormat, key and value class in configuration
+ *   myHadoopConfiguration.setClass("mapreduce.job.inputformat.class", InputFormatClass, InputFormat.class);
+ *   myHadoopConfiguration.setClass("key.class", InputFormatKeyClass, Object.class);
+ *   myHadoopConfiguration.setClass("value.class", InputFormatValueClass, Object.class);
+ * </tt>
+ * </pre>
+ * <p>
+ * Key/value translation is a mechanism provided to user for translating input format's key/value
+ * <br>
+ * type to any other class type for which a coder should be available.<br>
+ * For example:
+ * 
+ * <pre>
+ * <tt>
+ *  SimpleFunction<InputFormatKeyClass, MyKeyTranslate> myKeyTranslateFunc = new SimpleFunction<InputFormatKeyClass, MyKeyTranslate>() {
+ *    <tt>@Override</tt>
+ *     public MyKeyTranslate apply(InputFormatKeyClass input) {
+ *       //...logic to transform InputFormatKeyClass to MyKeyTranslate
+ *     }
+ *  };
+ *  
+ *  SimpleFunction<InputFormatValueClass, MyValueTranslate> myValueTranslateFunc = new SimpleFunction<InputFormatValueClass, MyValueTranslate>() {
+ *    <tt>@Override</tt>
+ *     public MyValueTranslate apply(InputFormatValueClass input) {
+ *       //...logic to transform InputFormatValueClass to MyValueTranslate
+ *     }
+ *  };   
+ * </tt>
+ * </pre>
+ * 
+ * <h3>Read configuration options supported for HadoopInputFormatIO</h3>
+ * <p>
+ * HadoopConfiguration is set using {@link #withConfiguration}. Key/value translation function<br>
+ * is optional and can be set using {@link #withKeyTranslation}/{@link #withValueTranslation}.<br>
+ * For example :
+ * 
+ * <pre>
```

Comment:
for your code snippets here, you can include the whole section in <pre> and then avoid doing any html formatting inside of them. That makes it much more readable when you're just looking at the java comment, and still works if HTML formatted. 

See the example at https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java#L88